### PR TITLE
ZCS-1797 :- Universal UI - Integrate Progress bar with Task list, list view - right view

### PR DESCRIPTION
**Pull request by SatyamKumarOnline**
Desc :- Changed the  underlying layout of the task list Items, to accommodate the circular progress task bar .

Changeset :-
ZmTaskListView.js :- Layout changed for list item.
zm.css :- Styles for circular progress task bar for list item.
skin.properties .

### CODE REVIEW

File: WebRoot/css/zm.css [hunk 1]
```diff
@@ -2014,6 +2006,20 @@ div.CompDragArea * {
...
+.ZmTaskMultiView .DwtListView  .RowDouble .ZmTaskHeaderImage .svg-icon,
+.ZmTaskMultiView .DwtListView  .RowDouble  .ZmTaskHeaderImage .progress-circle {
+	@TaskReadingPaneRight-ProgressBarImage@
+}
+
+.ZmTaskMultiView .DwtListView  .RowDouble .BottomRow {
+	@TaskReadingPaneRight-BottomRowIconContainer@
+}
+
+
```
Comment (tarangkhandelwal) [inline WebRoot/css/zm.css]: extra space here

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 2]
```diff
@@ -482,39 +482,36 @@ function(task, colIdx) {
...
-        idx = this._getAbridgedCell(htmlArr, idx, task, ZmItem.F_ATTACHMENT, colIdx, width);
-    }
-    htmlArr[idx++] = "</tr></table></td>";
-    htmlArr[idx++] = "</tr></table>";
+	if (task.hasAttach) {
+		idx = this._getAbridgedCell(htmlArr, idx, task, ZmItem.F_ATTACHMENT, colIdx, width);
+	}
 
+	htmlArr[idx++] = "</tr></table></td>";
+	htmlArr[idx++] = "</tr></table></td>";
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Previously there was only two closing for table tag. (</table>), now i see 3. Have I missed any new addition of table wrapper??

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 3]
```diff
@@ -624,7 +621,8 @@ function(el) {
 		var parentEl = document.getElementById(this._htmlElId);
 		parentEl.appendChild(this._newTaskInputEl);
 
-		this._newTaskInputEl.style.padding  = "0px";
+		this._newTaskInputEl.style.padding      = "0px";
+		this._newTaskInputEl.style.paddingLeft  = "14px";
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: should be via css.

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 2 above)
Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: See line `498` and `500`. Is it what you are asking for?

Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Not really, I mean that  there were two closings for table tag before and now I see 3 closings. I don't see any new addition of new table tag opening. Same with <tr> , I believe there are 2 rows (topRow & bottomRow) and I see  3 closing for that.
Correct me if I misunderstood something here.

Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Other table declaration is at line number: 485.

Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: oh Ok. Then it looks ok. Thanks for the explanation.

File: WebRoot/css/zm.css (hunk 1 above)
Comment (SatyamKumarOnline) [inline WebRoot/css/zm.css]: Removed

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 3 above)
Comment (SatyamKumarOnline) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Moved them to css files

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 4]
```diff
@@ -482,39 +482,36 @@ function(task, colIdx) {
...
 	// first row
-	htmlArr[idx++] = "<table width=100% class='TopRow'>";
+	htmlArr[idx++] = "<table class='TopRow'>";
 	htmlArr[idx++] = "<tr id='";
 	htmlArr[idx++] = DwtId.getListViewItemId(DwtId.WIDGET_ITEM_FIELD, this._view, task.id, ZmItem.F_ITEM_ROW_3PANE);
 	htmlArr[idx++] = "'>";
 
-    idx = this._getAbridgedCell(htmlArr, idx, task, ZmItem.F_SUBJECT, colIdx);
+	if (task.pComplete) {
+		htmlArr[idx++] = "<td class='ZmTaskHeaderImage'>";
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: To me this class name sounds little odd. Why are we using Header word here in the list view? Makes more sense to have it in detail view where we actually have a header. 
Can we update the name?

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 3 above)
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Not able to locate the change, I think the previous commit is overridden. Can you point me to the css change for this ? 
Thanks.

Comment (SatyamKumarOnline) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Has been moved to property TaskListViewNewTaskInputWidget

Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: oh ok. Thanks. 
Any specific reason why we increased the specificity for the selector? Just curious to know about that.

Comment (SatyamKumarOnline) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: There are also other generic selectors like INPUT[type="text"] , that was getting precedence over for common properties. 
To override the generic selector, increased the specificity for the selector   .

File: WebRoot/skins/_base/base4/skin.properties [hunk 5]
```diff
@@ -1468,15 +1468,18 @@ TaskInfo                = @ContactInfo@;
 
 # Task List view
 TaskListViewAddNewTaskIcon                                  = display:inline-block; margin-right:12px;
+TaskListViewNewTaskInputWidget                              = z-index:300; overflow:hidden; padding:0px 0px 0px 14px; border-width:0px;
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: why do we need z-index here?

Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: It was earlier used in the class, InlineWidget . 
I moved them to properties file

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 3 above)
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: okay. This is fine in that case. 👍

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 4 above)
Comment (SatyamKumarOnline) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: The selector 

`.ZmTaskHeaderImage .svg-icon,
.ZmTaskHeaderImage .progress-circle`

defines a set of properties for circular task progress bar used in detailed view.
For circular task progress bar, used in list view, all the properties are same but height and width.
That has been overridden in 
`.ZmTaskMultiView .DwtListView  .RowDouble .ZmTaskHeaderImage .svg-icon,
`.
To  reuse the existing property for circular task progress bar defined in detailed view, named class to ZmTaskHeaderImage.

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js [hunk 6]
```diff
@@ -482,39 +482,36 @@ function(task, colIdx) {
...
 
 	// first row
-	htmlArr[idx++] = "<table width=100% class='TopRow'>";
+	htmlArr[idx++] = "<table class='TopRow'>";
 	htmlArr[idx++] = "<tr id='";
 	htmlArr[idx++] = DwtId.getListViewItemId(DwtId.WIDGET_ITEM_FIELD, this._view, task.id, ZmItem.F_ITEM_ROW_3PANE);
 	htmlArr[idx++] = "'>";
 
-    idx = this._getAbridgedCell(htmlArr, idx, task, ZmItem.F_SUBJECT, colIdx);
+	if (task.pComplete) {
```
Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: This condition will not render progress bar for task with `0%` progress. We however need to show such progress-bar as well. https://zpl.io/1JJMkw

Thanks

Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: I would suggest, we test it for 0 & 100% both as the behaviour is little different for those cases.

File: WebRoot/skins/_base/base4/skin.properties (hunk 5 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.properties]: But I think it should work without `z-index` as well. Can you please check and if it's working then I think we should remove `z-index`.

Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: Removed z-index.

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 6 above)
Comment (SatyamKumarOnline) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: Updated conditions to work for both 0 and 100 .

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 4 above)
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: There are some css rules already present with that class name (zm.css near line 2358):
```
.ZmTaskProgress {
	@FullWidth@
}

.ZmTaskProgress DIV {
	display:block;
	height:100%;
}
```
Do we need these now?

File: WebRoot/skins/_base/base4/skin.properties (hunk 5 above)
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: thanks

File: WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js (hunk 4 above)
Comment (SatyamKumarOnline) [inline WebRoot/js/zimbraMail/tasks/view/ZmTaskListView.js]: This is obsolete now.
So, removed them.
Thanks .

**Pull request merged**